src/Mod/Mesh/Gui/SegmentationBestFit.cpp

```diff
@@ -505,17 +505,15 @@ void SegmentationBestFit::accept()
 
     std::string internalname = "Segments_";
     internalname += myMesh->getNameInDocument();
-    App::DocumentObjectGroup* group = static_cast<App::DocumentObjectGroup*>(
-        document->addObject("App::DocumentObjectGroup", internalname.c_str()));
+    auto* group = document->addObject<App::DocumentObjectGroup>(internalname.c_str());
```

Comment:
Something to add to the guidelines I think. We use both `auto` and `auto*` for pointers, we should settle on one way of writing this.

Reply:
Though, some of the changes made in this pr keeps the way it was written before